content/en/docs/armory-agent/agent-mtls.md

```diff
@@ -0,0 +1,262 @@
+      clouddriver:
+        # See https://docs.armory.io/docs/installation/armory-agent/agent-options/
+        kubesvc:
+          cluster: redis
+          grpc:
+            server:
+              security:
+                enabled: true
+                trustCertCollection: file:<path-to-ca.crt>
+                certificateChain: file:<path-to-clouddriver.crt>
+                privateKey: file:<path-to-clouddriver.key>
+                clientAuth: REQUIRE
+{{< /prism >}}
+
+See the {{< linkWithTitle "agent-plugin-options.md" >}} page for additional options.
+
+## Agent service configuration
+
+### Create a secret
+
+Create a secret in the target cluster namespace where the Agent resides.
+
+```bash
+kubectl create secret generic <agent-secret-name> \
+--from-file=agent.crt --from-file=agent.key
+```
+
+### Modify deployment configuration
+
+Modify the Agent's deployment configuration in `deployment.yaml` to mount the certs.
+
+If you have a custom CA, you need to mount the cert into the known trusted cert location:  `/etc/ssl/cert.pem`. Golang apps in Alpine use `/etc/ssl/cert.pem` as the source of a trusted CA.
+
+  1. Obtain a copy of the `cert.pem` file from a running Agent.
+  2. Append your custom CA onto the copied `cert.pem`  (for example `cat ca.crt >> cert.pem`)
+  3. Mount `cert.pem` to Agent deployment
+
+>The paths that files are mounted to in the `deployment.yaml` file should always match the corresponding location in the `kubesvc.yaml` configuration file. For example, the `mountPath` of the CA cert in the `deployment.yaml` file must match the `clouddriver.tls.clientCertFile` location in `kubesvc.yaml`.
+
+{{< prism lang="yaml"  >}}
+spec:
+  template:
+    spec:
+      containers:
+        volumeMounts:
+        - mountPath: <path> # for example, /opt/kubesvc/cert
+          name: armoryagentcert
+        - mountPath: <path> # for example, /opt/kubesvc/cacert
+          name: clouddrivercacert         
+      volumes:
+      - name: armoryagentcert
+        secret:
+          secretName: <agent-secret-name>
+      - name: clouddrivercacert         
+        secret:                         
+          secretName: <clouddriver-secret-name>
+      - name: certpem
+        secret:
+          secretName: <ca-secret-name>
+{{< /prism >}}
+
+
+### Configure the service
+
+Add the certificate information in `kubesvc.yaml`:
+
+{{< prism lang="yaml" >}}
+clouddriver:
+  grpc: <:443
+  insecure: false
+  tls:
+    #serverName: <my-ca> #to override the server name of the certificate authority
+    insecureSkipVerify: false #if true, don't verify server's cert
+    clientKeyFile: <path>/agent.key #ref to the private key (mTLS)
+    #clientKeyPassword: #if the clientKeyFile is password protected
+    #cacertFile: <path>/ca.crt #to validate server's cert
+    clientCertFile: <path>/agent.crt #client cert for mTLS.
+
+{{< /prism >}}
+
+See the {{< linkWithTitle "agent-options.md" >}} for additional options.
+
+## Forward proxy configuration
```

Comment:
As a reader, this section leaves me confused. Do I need a forward proxy? What is the forward proxy for? When would I need it? What problem is it solving?

Reply:
I'm more confused than you are.

Reply:
I removed it because it's got nothing to do with mTLS.